esp/src/eclwatch/DynamicESDLQueryWidget.js

```diff
@@ -172,27 +172,28 @@ define([
                 }
             }, domID);
 
-            retVal.on(".dgrid-row-url:click", function (evt) {
-                if (context._onRowDblClick) {
-                    var item = retVal.row(evt).data;
-                    context._onRowDblClick(item);
-                }
-            });
+            ESPUtil.goToPageUserPreference(retVal, "DynamicESDLQueryWidget_GridRowsPerPage").then(function(){
+                retVal.on(".dgrid-row-url:click", function (evt) {
+                    if (context._onRowDblClick) {
+                        var item = retVal.row(evt).data;
+                        context._onRowDblClick(item);
+                    }
+                });
 
-            retVal.on(".dgrid-row:dblclick", function (evt) {
-                evt.preventDefault();
-                context.grid.refresh()
-            });
+                retVal.on(".dgrid-row:dblclick", function (evt) {
+                    evt.preventDefault();
+                    context.grid.refresh()
+                });
 
-            retVal.onSelectionChanged(function (event) {
-                var selection = retVal.getSelected();
-                if (selection.length > 0) {
-                    context.deleteButton.set("disabled", false);
-                } else {
-                    context.deleteButton.set("disabled", true);
-                }
+                retVal.onSelectionChanged(function (event) {
+                    var selection = retVal.getSelected();
+                    if (selection.length > 0) {
+                        context.deleteButton.set("disabled", false);
+                    } else {
+                        context.deleteButton.set("disabled", true);
+                    }
+                });
```

Comment:
There is no need to put the "on" calls in the "then" - its that startup thats important.